lib/server.js

```diff
@@ -254,6 +255,19 @@ module.exports = class Server extends EventEmitter {
       return hs
     }
 
+    if (this._shareLocalAddress && !direct && clientAddress.host === serverAddress.host) {
```

Comment:
This added condition of `_shareLocalAddress` fixes two failing tests, but just realized it's wrong because server might still want to do local connections or at least it's still capable of doing so, even if it's not sharing local addresses